pkg/utils/value/value.go

```diff
@@ -0,0 +1,112 @@
+package value
+
+import (
+	"fmt"
+	"strings"
+)
+
+type MapValue map[string]interface{}
+
+type Setter func(v interface{})
+
+type MapValueCallback func(setter Setter, v interface{}, e Extra) error
```

Comment:
- it would be better to use pointer(*Extra) instead of struct for `Extra`, the latter will cause a struct copy.
- we can move `setter` to `Extra`, make callback function more simple